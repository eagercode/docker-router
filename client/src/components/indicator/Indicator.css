.Indicator:before {
    content: ' \25CF';
    float: right;
    font-size: 1.5em;
    right: 3%;
    position: absolute;
    top: 0.2em;
}

.Indicator.Indicator-green {
    color: #4CAF50
}

.Indicator.Indicator-red {
    color: #E53935
}
