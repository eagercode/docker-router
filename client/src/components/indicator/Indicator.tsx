import * as React from 'react';
import Span from './Indicator.sc';

export interface Props {

    isActive: boolean;
}

export default class Indicator extends React.Component<Props, {}> {

    render(): JSX.Element {
        return (
            <Span isActive={this.props.isActive}/>
        );
    }
}
