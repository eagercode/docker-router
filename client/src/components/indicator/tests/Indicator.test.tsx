import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Indicator from '../Indicator';

describe('<Indicator/>', () => {
    it('active indicator snapshot', () => {
        expect(mount(<Indicator isActive={true}/>)).toMatchSnapshot();
    });

    it('inactive indicator snapshot', () => {
        expect(mount(<Indicator isActive={false}/>)).toMatchSnapshot();
    });
});