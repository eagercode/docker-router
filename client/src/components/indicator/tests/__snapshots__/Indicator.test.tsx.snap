// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Indicator/> active indicator snapshot 1`] = `
.c0 {
  color: #4CAF50;
}

.c0:before {
  content: " \\25CF";
  float: right;
  font-size: 1.5em;
  right: 3%;
  position: absolute;
  top: 0.2em;
}

<Indicator
  isActive={true}
>
  <styled.span
    isActive={true}
  >
    <span
      className="c0"
    />
  </styled.span>
</Indicator>
`;

exports[`<Indicator/> inactive indicator snapshot 1`] = `
.c0 {
  color: #E53935;
}

.c0:before {
  content: " \\25CF";
  float: right;
  font-size: 1.5em;
  right: 3%;
  position: absolute;
  top: 0.2em;
}

<Indicator
  isActive={false}
>
  <styled.span
    isActive={false}
  >
    <span
      className="c0"
    />
  </styled.span>
</Indicator>
`;
